code/modules/mining/boulder_processing/_boulder_processing.dm

```diff
@@ -0,0 +1,344 @@
+	. = ..()
+	if(anchored)
+		icon_state ="[initial(icon_state)]"
+	else
+		icon_state ="[initial(icon_state)]-off"
+
+/obj/machinery/bouldertech/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_unfasten_wrench(user, tool, time = 1.5 SECONDS) == SUCCESSFUL_UNFASTEN)
+		update_appearance(UPDATE_ICON_STATE)
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bouldertech/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_screwdriver(user, "[initial(icon_state)]-off", initial(icon_state), tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bouldertech/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_pry_open(tool, close_after_pry = TRUE, closed_density = FALSE))
+		return ITEM_INTERACT_SUCCESS
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bouldertech/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(.)
+		return
+	if(holds_minerals && istype(attacking_item, /obj/item/boulder))
+		var/obj/item/boulder/my_boulder = attacking_item
+		update_boulder_count()
+		if(!accept_boulder(my_boulder))
+			balloon_alert_to_viewers("full!")
+			return
+		balloon_alert_to_viewers("accepted")
+		START_PROCESSING(SSmachines, src)
+		return TRUE
+	if(!istype(attacking_item, /obj/item/card/id))
+		return
+	if(!holds_mining_points)
+		return FALSE
+	if(points_held <= 0)
+		balloon_alert_to_viewers("no points to claim!")
+		if(!COOLDOWN_FINISHED(src, sound_cooldown))
+			return TRUE
+		COOLDOWN_START(src, sound_cooldown, 1.5 SECONDS)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, FALSE)
+		return FALSE
+	var/obj/item/card/id/id_card = attacking_item
+	var/amount = tgui_input_number(user, "How many mining points do you wish to claim? ID Balance: [id_card.registered_account.mining_points], stored mining points: [points_held]", "Transfer Points", max_value = points_held, min_value = 0, round_value = 1)
+	if(!amount)
+		return
+	if(amount > points_held)
+		amount = points_held
+	id_card.registered_account.mining_points += amount
+	points_held = round(points_held - amount)
+	to_chat(user, span_notice("You claim [amount] mining points from \the [src] to [id_card]."))
+	return TRUE
+
+/obj/machinery/bouldertech/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!anchored)
+		balloon_alert(user, "anchor first!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(!holds_minerals)
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	if(!length(boulders_contained))
+		balloon_alert_to_viewers("No boulders to remove!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	remove_boulder(pick(boulders_contained))
+	return SECONDARY_ATTACK_CONTINUE_CHAIN
+
+/obj/machinery/bouldertech/deconstruct(disassembled)
+	. = ..()
+	if(length(contents))
+		for(var/obj/item/boulder/boulder in contents)
+			remove_boulder(boulder)
+
+/obj/machinery/bouldertech/process()
+	if(!anchored)
+		return
```

Comment:
Not mandatory but rather than a no-op you can start processing when it is anchored & stop processing when unanchored